feat(uitests): add support for iOS dynamic type settings in UITests

NicFontana opened the pull request:
<!--- IMPORTANT: Please review [how to contribute](../CONTRIBUTING.md) before proceeding further. -->
<!--- IMPORTANT: If this is a Work in Progress PR, please mark it as such in GitHub. -->

## Description
The solution I'm proposing is a little hacky, but it should help running UITests with different font sizes without manually change the setting of the simulators.
It is based on the undocumented parameters I've found [here](https://github.com/fastlane/fastlane/issues/2490#issuecomment-193498984).

## Checklist

<!--- Please insert an ‘x’ after you complete each step -->

- [x] I have followed the indications in the [CONTRIBUTING](../CONTRIBUTING.md).
- [x] The documentation related to the proposed change has been updated accordingly (plus comments in code).
- [x] I have written new tests for my core changes, as applicable.
- [x] I have successfully run tests with my changes locally.
- [x] It is ready for review! :rocket:

## Fixes

<!-- Please insert the issue numbers after the # symbol -->

- Fixes #40

NicFontana committed:
feat(uitests): add support for iOS dynamic type settings in UITests

NicFontana committed:
style: fix a typo

ImmuniDGR commented:
🚔 **Safety Check** 🚔

🔰 **Result** 🔰


✅ All configuration files are in line with the master branch.

🛠 **Diagnostic information** 🛠
- CircleCI scheduler pipeline: #392 (id: 46582238-07bb-4997-941e-67fea2004b56)
- Last verified commit: #986d5804bc921bbd13e691443f0ab91482426933
- Time of check: 27/05/2020, 17:16:28 UTC
- The following protected files have been checked for changes: CI/Export/development.plist, swiftgen.yml, dangerfile.ts, CI/danger/swiftformat.ts, Brewfile, Makefile, fastlane/Fastfile, Podfile, project.yml, fastlane/Appfile, project_debug.yml, scheduler_config.json, .swiftformat, .swiftlint.yml, .gitmodules, CI/danger/utils.ts, CI/changelog_config.tpl, CI/Export/app-store.plist, CI/danger/commitlint/rules.ts, yarn.lock, Podfile.lock, CI/danger/danger.ts, package.json, CI/danger/swiftlint.ts, publiccode.yml, CI/danger/commitlint/index.ts.

NicFontana committed:
fix: fix run.py arguments length check

Also ignore "xcrun simctl shutdown <device>" command's error raised when a device is already turned off.

NicFontana commented on UITests/run.py:
```diff
@@ -61,8 +85,37 @@
+		# set the simulator's font size
+		print("Set the {} simulator\'s font size to {}".format(device, font_size))
+		simulator_largerText_preferences_file = '{}/Library/Developer/CoreSimulator/Devices/{}/data/Library/Preferences/com.apple.preferences-framework.plist'.format(os.path.expanduser("~"), simulator_uuid)
+		plistlib.writePlist({'largeTextUsesExtendedRange': True}, simulator_largerText_preferences_file) # enable larger text
```
```plistlib.writePlist()``` is [deprecated](https://docs.python.org/3/library/plistlib.html#plistlib.writePlist) since python 3.4. I'm using it for compatibility with older versions since it is not specified a python version required to run this script.

NicFontana commented on UITests/run.py:
```diff
@@ -61,8 +85,37 @@
+		plistlib.writePlist({'largeTextUsesExtendedRange': True}, simulator_largerText_preferences_file) # enable larger text
+
+		subprocess.call(['plutil', '-replace', 'UIPreferredContentSizeCategoryName',
+		'-string', font_size, simulator_fontSize_preferences_file]) # set the size
```
This can be accomplished with ```plistlib``` but sometimes I had problems with the process of reading and writing back the plist file using the deprecated APIs. Instead, in this way, I have never had problems.

ImmuniDGR commented:
🚔 **Safety Check** 🚔

🔰 **Result** 🔰
<ul>
<li>The CircleCI configuration file has been modified.</li>
</ul>

⚠️ Some configuration files don't match the master branch.

🛠 **Diagnostic information** 🛠
- CircleCI scheduler pipeline: #418 (id: 2eaf1d0f-181a-4d39-bdf5-cc72db7c4dcb)
- Last verified commit: #c95e6679ea98e8e1c43e4ed69b7cdecd445bde75
- Time of check: 27/05/2020, 20:45:50 UTC
- The following protected files have been checked for changes: CI/changelog_config.tpl, Podfile, package.json, project.yml, scheduler_config.json, .swiftlint.yml, CI/danger/utils.ts, .gitmodules, CI/danger/commitlint/rules.ts, fastlane/Fastfile, CI/Export/app-store.plist, yarn.lock, dangerfile.ts, project_debug.yml, CI/danger/commitlint/index.ts, Podfile.lock, swiftgen.yml, Makefile, fastlane/Appfile, .swiftformat, CI/danger/swiftlint.ts, CI/danger/danger.ts, publiccode.yml, Brewfile, CI/danger/swiftformat.ts, CI/Export/development.plist.

NicFontana commented on UITests/run.py:
```diff
@@ -61,8 +85,37 @@
+		print("Boot the {} simulator".format(device))
+		subprocess.call(["xcrun", "simctl", "boot", device])
+	else:
+		print("Unable to set font size on the {} simulator. This happen if it's the first time you boot it, try it again.".format(device))
```
Unfortunately this could happen because the file needed to change the simulator's font size: ```~/Library/Developer/CoreSimulator/Devices/<device uuid>/data/Library/Preferences/com.apple.UIKit.plist```
is generated the first time the simulator is booted.

NicFontana commented on UITests/run.py:
```diff
@@ -61,8 +85,37 @@
+
+		# set the simulator's font size
+		print("Set the {} simulator\'s font size to {}".format(device, font_size))
+		simulator_largerText_preferences_file = '{}/Library/Developer/CoreSimulator/Devices/{}/data/Library/Preferences/com.apple.preferences-framework.plist'.format(os.path.expanduser("~"), simulator_uuid)
```
This file is not present until we manually enable the ```Larger Accessibility Sizes``` setting.
Anyway, this will be edited if present or created if not.

bolismauro reviewed (approved):
Thanks! Looks great :)

NicFontana commented:
> Thanks! Looks great :)

Glad I can help :)
